File: app/src/main/java/jp/hazuki/yuzubrowser/action/view/ActionListViewAdapter.java
package jp.hazuki.yuzubrowser.action.view;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import jp.hazuki.yuzubrowser.action.Action;
import jp.hazuki.yuzubrowser.action.ActionList;
import jp.hazuki.yuzubrowser.action.ActionNameArray;

public class ActionListViewAdapter extends ArrayAdapter<Action> {
    private final ActionNameArray mActionNameArray;

    public ActionListViewAdapter(Context context, ActionList objects, ActionNameArray array) {
        super(context, 0, objects);
        mActionNameArray = (array != null) ? array : new ActionNameArray(context);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(android.R.layout.simple_list_item_1, null);
        }
        Action action = getItem(position);
        if (action != null) {
            ((TextView) convertView.findViewById(android.R.id.text1)).setText(action.toString(mActionNameArray));
        }
        return convertView;
    }
}

File: legacy/src/main/java/jp/hazuki/yuzubrowser/legacy/action/view/ActionListViewAdapter.kt
/*
 * Copyright (C) 2017 Hazuki
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jp.hazuki.yuzubrowser.legacy.action.view

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import jp.hazuki.yuzubrowser.legacy.action.Action
import jp.hazuki.yuzubrowser.legacy.action.ActionList
import jp.hazuki.yuzubrowser.legacy.action.ActionNameArray

class ActionListViewAdapter(context: Context, objects: ActionList, array: ActionNameArray?) : ArrayAdapter<Action>(context, 0, objects) {
    private val mActionNameArray = array ?: ActionNameArray(context)

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view = convertView ?: LayoutInflater.from(context).inflate(android.R.layout.simple_list_item_1, null)

        getItem(position)?.let {
            view.findViewById<TextView>(android.R.id.text1).text = it.toString(mActionNameArray)
        }
        return view
    }
}
